cpp/modmesh/math.hpp

```diff
@@ -26,6 +26,79 @@ inline constexpr T pow(T base, std::integral_constant<size_t, N> /*unused*/)
     return pow(base, std::integral_constant<size_t, N - 1>()) * base;
 }
 
+template <typename T>
+inline constexpr T pi_v = std::enable_if_t<std::is_floating_point_v<T>, T>(3.141592653589793238462643383279502884L);
```

Comment:
This gives us a reason to move on to C++20: http://eel.is/c++draft/numbers

Reply:
Do you need the `inline`?

Reply:
I learned from the https://en.cppreference.com/w/cpp/language/inline that `constexpr` variables with external linkage or in multiple translation units should have `inline` to avoid multiple definition error.
Although the cppreference does not mention whether template variables need inline, but I guessed it needed, so I added it.

Reply:
Could you please try without `inline` here?  I think a template does not need it.  If it works without `inline`, it's better to avoid it for consistency.

Reply:
I have tested it without `inline` and it is working, therefore I have removed it in  https://github.com/solvcon/modmesh/pull/441/commits/5b17c51fb0ae9e67e692e8dc5576b5328336c795